src/lib/components/NglMoleculeViewer.react.js

```diff
@@ -0,0 +1,193 @@
+import React, {Component, lazy, Suspense} from 'react';
+import LazyLoader from '../LazyLoader';
+import PropTypes from 'prop-types';
+
+const RealNglMoleculeViewer = lazy(LazyLoader.nglmoleculeViewer);
+/**
+ * The NglMoleculeViewer is used to render schematic diagrams
+ * of biomolecules in ribbon-structure representations.
+ * Read more about the component here:
+ * https://github.com/IvoLeist/dash_ngl
+ * Read more about the used WebGL protein viewer here:
+ * https://github.com/arose/ngl
+ */
+export default class NglMoleculeViewer extends Component {
+    render() {
+        return (
+            <Suspense fallback={null}>
+                <RealNglMoleculeViewer {...this.props} />
+            </Suspense>
+        );
+    }
+}
+
+const defaultViewportStyle = {
+    width: '500px',
+    height: '500px',
+};
+
+const defaultStageParameters = {
+    quality: 'medium',
+    backgroundColor: 'white',
+    cameraType: 'perspective',
+};
+
+const defaultImageParameters = {
+    antialias: true,
+    transparent: true,
+    trim: true,
+    defaultFilename: 'dash-bio_ngl_output',
+};
+
+const defaultData = [
+    {
+        filename: 'placeholder',
+        ext: '',
+        selectedValue: 'placeholder',
+        chain: 'ALL',
+        aaRange: 'ALL',
+        chosen: {
+            chosenAtoms: '',
+            chosenResidues: '',
+        },
+        color: 'red',
+        config: {
+            input: '',
+            type: 'text/plain',
+        },
+        uploaded: false,
+        resetView: false,
+    },
+];
+
+NglMoleculeViewer.defaultProps = {
+    data: defaultData,
+    viewportStyle: defaultViewportStyle,
+    stageParameters: defaultStageParameters,
+    imageParameters: defaultImageParameters,
+    downloadImage: false,
+    molStyles: {
+        representations: ['cartoon', 'axes+box'],
+        chosenAtomsColor: '#ffffff',
+        chosenAtomsRadius: 1,
+        molSpacingXaxis: 100,
+        sideByside: false,
+    },
+};
+
+NglMoleculeViewer.propTypes = {
+    /**
+     * The ID of this component, used to identify dash components in callbacks.
+     * The ID needs to be unique across all of the components in an app.
+     */
+    id: PropTypes.string,
+
+    /**
+     * The height and the width (in px) of the container
+     * in which the molecules will be displayed.
+     * It should be in JSON format.
+     */
+    viewportStyle: PropTypes.exact({
+        width: PropTypes.string,
+        height: PropTypes.string,
+    }),
```

Comment:
Can we break this out into two props, `width` and `height`, each of which is just a number? The current implementation expects a CSS string, which is indeed more flexible but it's harder to use for non-web-devs. If you think this flexibility is important we could support both, like `widthStr = isNumeric(width) ? (width + 'px') : width` (using my favorite [`fast-isnumeric`](https://www.npmjs.com/package/fast-isnumeric) 😉 so `500` and `'500'` count as numbers, ie pixels, but any other string is a CSS size string) or we can start with just numbers and include strings later if there's interest.

Reply:
Added in 78fd7e9.